bmc/virtual_media.go

```diff
@@ -0,0 +1,70 @@
+package bmc
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/pkg/errors"
+)
+
+// VirtualMediaSetter controls the virtual media attached to a machine
+type VirtualMediaSetter interface {
+	SetVirtualMedia(ctx context.Context, kind string, mediaUrl string) (ok bool, err error)
+}
+
+// VirtualMediaProviders is an internal struct to correlate an implementation/provider and its name
+type virtualMediaProviders struct {
+	name               string
+	virtualMediaSetter VirtualMediaSetter
+}
+
+// setVirtualMedia sets the virtual media.
+func setVirtualMedia(ctx context.Context, kind string, mediaUrl string, b []virtualMediaProviders) (ok bool, metadata Metadata, err error) {
```

Comment:
can we have tests for the methods in this file, hoping at some point we could generate the boilerplate code here along with tests.